spec/rspec/mocks/matchers/receive_spec.rb

```diff
@@ -109,6 +123,18 @@ module Mocks
 
               expect(receiver.foo(kw: :arg)).to eq(:arg)
             end
+
+            it "expects to receive keyword args" do
+              dbl = instance_double(TestObject)
+              expect(dbl).to receive(:kw_args_method).with(a: 1, b: 2)
+              dbl.kw_args_method(a: 1, b: 2)
+            end
+
+            it "expects to receive keyword args with a hash" do
+              dbl = instance_double(TestObject)
+              expect(dbl).to receive(:kw_args_method).with({a: 1, b: 2})
+              dbl.kw_args_method(a: 1, b: 2)
```

Comment:
Isn't this expected to fail? It expects a positional Hash but is given kwargs.
But I guess `expect(dbl).to receive` actually creates a method with no kwargs so then it should pass?

Reply:
This indeed should have failed on earlier Ruby versions, too.